interview-questions.md

```diff
@@ -8,42 +8,42 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: I cannot recall object- oriented programming. I do know however that the coding language utilizes it. It is because the data types in ruby are all separated into different class types. Therefore any function that is made for an object in a class would be called a method. It is utilized so objects are the focus of manipulation instead of just the basic datatypes by themselves. Functional programming is based on the use of functions themselves in their usuability not applied to a specific object. Therefore functional programming is more universal while object oriented programming requires a known value to be utilized.
 
-Researched answer:
+Researched answer:Object-oriented programming is more focused on the manipulation of objects rather than functions and logic. It is used more for large and complex programs with a large collaborative effort. An issue with that is that an object becomes a mutable or non changing data. Functional programming focuses on the logic in which anything can be passed into it and an outcome is achieved. Functions can be inputed in other functions to create higher order functions allowing for more complex processes to be made.
 
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: The difference between float and an integer resides in the decimal point. Integers are the whole numbers such as 1,2,3. A float is the place after the decimal point is laid out and a number is added to the right of it. For an example, 1.0, 2.0, and 3.0 would all be float. 
 
-Researched answer:
+Researched answer: They are both numerical datatypes. The difference is that float contains a decimal point allowing for more realistic calculations while integers cannot. For an example, we could do 5/2, the outcome in integers would be 2 rounding to the closest int. In order for us to get a float answer we would need either 5 and 2 to be 5.0 or 2.0. This would lead to an exact answer of 2.5.
 
 3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: Implicit return is the act of a function to output the last line of a function without needing a return to state as such. It allows for ease of typing out functions for ruby since the last line of a function is usually what is required from it.
 
-Researched answer:
+Researched answer: Implicit return will output the last line of code prior to the end keyword. Usually a return keyword will be used to signal what to output from the function however sometimes conditions in the function make the return keyword not useful. This is because the return keyword will stop all other processes in the function from occurring once it is read.
```

Comment:
Great stuff! How is this solution to the return better? What implications does it have?